blivet/devices/disk.py

```diff
@@ -725,3 +728,101 @@ def description(self):
     @property
     def sector_size(self):
         return self._sector_size
+
+
+NVMeController = namedtuple("NVMeController", ["name", "serial", "nvme_ver", "id", "subsysnqn"])
+
+
+class NVMeNamespaceDevice(DiskDevice):
```

Comment:
It would be nice if this could be called `NVMeDevice` and so on for the other device and the populators, but I wouldn't want it to be confusing or vague if that's a possibility.

Reply:
Strictly speaking the device is a namespace device (the controller is the "nvme device") and there can be multiple namespaces on one device. And we already have `NVDIMMNamespaceDevice` for nvdimms. But we are not going to have a device for the character device controller so there shouldn't be a conflict if we decide to name the namespaces simply `NVMeDevice`. 
Any thoughts about the naming here @tbzatek ?

Reply:
From my point of view the plain `NVMeDevice` is somewhat misleading. While at the moment only namespaces are of interest, certain operations need to be performed on a controller device. Modelling both namespace and controller devices separately is more future-proof. Of course, depends on the future plans.

Reply:
Sounds like there are several reasons to leave it as-is.